File: lua/pe012.lua
-- this computes projecteuler.net problem 12

local function get_num_divisors(num)
  counter = 0
  running = 1
  while running <= math.sqrt(num) do
    if (num % running) == 0 then
      counter = 1 + counter
    end
    running = 1 + running
  end
  return counter
end

local last = 250
local num = 0
local i = 0
local j = 1

while num < last do
  i = i + j
  num = get_num_divisors(i)
  j = 1 + j
end

print(i)

File: python/pe012.py
#!/usr/bin/env python3

""" this computes projecteuler.net problem 12 """

import sys
from math import sqrt


def get_num_divisors(number):
    '''this function compute the number of divisors of the passed number'''
    counter = 0
    running = 1
    while running <= sqrt(number):
        if not number % running:
            counter += 1
        running += 1
    return counter


LAST = 250
num = 0
i = 0
j = 1

while num < LAST:
    i = i + j
    num = get_num_divisors(i)
    j += 1

print(i)

sys.exit(0)
